tests/readme.md

```diff
@@ -0,0 +1,77 @@
+# Testing
```

Comment:
Planning to add a link to this doc and some guidelines in the top-level readme, but waiting until we solidify some of these changes and process

Reply:
eventually in CONTRIBUTING.md